server/auth.go

```diff
@@ -78,15 +91,74 @@ func (h *Handler) LoginPost(c echo.Context) error {
 }
 
 func (h *Handler) LoginGet(c echo.Context) error {
+	sess, _ := session.Get(CookieKeySession, c)
+
 	vars := map[string]interface{}{
 		"csrf":               c.Get("csrf").(string),
 		"globus":             viper.GetBool("globus_signup"),
 		"enable_user_signup": viper.GetBool("enable_user_signup"),
+		"failure":            sess.Flashes("failure"),
+		"success":            sess.Flashes("success"),
 	}
 
+	sess.Save(c.Request(), c.Response())
+
 	return c.Render(http.StatusOK, "login.html", vars)
 }
 
+func (h *Handler) ChangePost(c echo.Context) error {
+	message := ""
+	sess, _ := session.Get(CookieKeySession, c)
+
+	expired := sess.Values["password-expired"]
+
+	if expired == nil || sess.Values["uid"] == nil || !expired.(bool) {
+		sess.AddFlash("You're not allowed to use this function directly!", "failure")
+		sess.Save(c.Request(), c.Response())
+		return c.Redirect(http.StatusFound, Path("/auth/login"))
```

Comment:
Let's return: `return echo.NewHTTPError(http.StatusUnauthorized, "Access denied here")`. No need for the flash message as the default 401 template will be displayed. Let's also do this for the GET request as well

Reply:
Good point. Definitely cleaner. Done.